From 97299e161049823ff13433254d8cf8a99b4bdf5b Mon Sep 17 00:00:00 2001
From: Binh Nguyen <binh.nguyen.jz@renesas.com>
Date: Mon, 22 Mar 2021 22:54:39 +0700
Subject: [PATCH 135/135] arm64: dts: rzg2-smarc: Add option build for
 CAN/ADC/GPT/SDHI1

This commit add some options build for JP/SW setting since
some modules have pin conflict:
- CAN <> ADC
- GPT <> SDHI1

Signed-off-by: Binh Nguyen <binh.nguyen.jz@renesas.com>
---
 arch/arm64/boot/dts/renesas/rzg2-smarc.dtsi | 34 +++++++++++++++++++++++++----
 1 file changed, 30 insertions(+), 4 deletions(-)

diff --git a/arch/arm64/boot/dts/renesas/rzg2-smarc.dtsi b/arch/arm64/boot/dts/renesas/rzg2-smarc.dtsi
index 9e1b901..7d6bbbf 100644
--- a/arch/arm64/boot/dts/renesas/rzg2-smarc.dtsi
+++ b/arch/arm64/boot/dts/renesas/rzg2-smarc.dtsi
@@ -23,6 +23,12 @@
  *	aplay -D hw:1,0 xxx.wav
  */
 
+/* Select one funtion support SDHI1|GPT (1: enabled, 0 : disabled)
+ * SDHI1 and GPT have pin conflict. Only one can be enabled at once
+ */
+#define SDHI1_SUPPORT		1
+#define GPT_SUPPORT		0
+
 /* Enable/Disable CAN support (must choose 1) */
 #define CAN_SUPPORT		1
 //#define CAN_SUPPORT		0
@@ -43,6 +49,10 @@
  */
 #define DISP_DSI_HDMI		1
 #define DISP_DSI_PANEL		0
+
+/* CAN and ADC have pin conflict. When Can is enabled, disable ADC
+ */
+#define ADC_SUPPORT		0
 #else
 /* Parallel display is supported as default if not support CAN.
  * Please choose one of two interfaces (HDMI or panel) based on your board
@@ -57,6 +67,10 @@
  */
 #define DISP_DSI_HDMI		0
 #define DISP_DSI_PANEL		0
+
+/* CAN and ADC have pin conflict. When Can is disabled, enable ADC
+ */
+#define ADC_SUPPORT		1
 #endif
 
 /* Below macros are created to check which video interface is supporting
@@ -273,6 +287,7 @@
 		};
 	};
 
+#if (SDHI1_SUPPORT)
 	sdhi1_pins: sd1 {
 		sd1_data {
 			pins =	"SD1_DATA0", "SD1_DATA1", "SD1_DATA2",
@@ -315,6 +330,7 @@
 		output-high;
 		line-name = "sd1_pwr_en";
 	};
+#endif
 
 	sound_clk_pins: sound_clk {
 		pins = "AUDIO_CLK1", "AUDIO_CLK2";
@@ -374,15 +390,19 @@
 		function = "rspi1";
 	};
 
+#if (GPT_SUPPORT)
 	gpt3_pins: gpt3 {
 		groups = "gpt3_b_b";
 		function = "gpt3";
 	};
+#endif
 
+#if (ADC_SUPPORT)
 	adc_pins: adc {
 		groups = "adc_d";
 		function = "adc";
 	};
+#endif
 
 	i2c1_pins: i2c1 {
 		pins = "RIIC1_SDA", "RIIC1_SCL";
@@ -460,6 +480,7 @@
 	status = "okay";
 };
 
+#if (SDHI1_SUPPORT)
 &sdhi1 {
 	pinctrl-0 = <&sdhi1_pins>;
 	pinctrl-1 = <&sdhi1_pins_uhs>;
@@ -473,6 +494,7 @@
 	sd-uhs-sdr104;
 	status = "okay";
 };
+#endif
 
 #if (CAN_SUPPORT)
 &can {
@@ -664,19 +686,23 @@
 	status = "okay";
 };
 
+#if (GPT_SUPPORT)
 &gpt3 {
 	pinctrl-0 = <&gpt3_pins>;
 	pinctrl-names = "default";
-	//conflict pin with SD1 WP, disable as default
-	//status = "okay";
+	//conflict pin with SD1 WP
+	status = "okay";
 };
+#endif
 
+#if (ADC_SUPPORT)
 &adc {
 	pinctrl-0 = <&adc_pins>;
 	pinctrl-names = "default";
-	//conflict pin with CAN0 pin, disable as default
-	//status = "okay";
+	//conflict pin with CAN0 pin
+	status = "okay";
 };
+#endif
 
 &i2c1 {
 	pinctrl-0 = <&i2c1_pins>;
-- 
2.7.4

